packages/wrangler/src/publish.ts

```diff
@@ -102,14 +102,36 @@ export default async function publish(props: Props): Promise<void> {
       }
     );
 
+    // Some validation of durable objects + migrations
+    if (config.durable_objects.bindings.length > 0) {
+      // TODO: implement durable objects for service environments
+      if (!props.legacyEnv) {
+        throw new Error(
+          "Publishing Durable Objects to a service environment is not currently supported. We'll fix this soon!"
+        );
+      }
+
+      // intrinsic [durable_objects] implies [migrations]
+      if (
+        config.durable_objects.bindings.filter(
+          (binding) => !binding.script_name
+        ).length > 0 &&
+        config.migrations.length === 0
+      ) {
+        console.warn(
+          "In wrangler.toml, you have configured intrinsic [durable objects], but no [migrations]. This may not work as expected until you add a [migrations] section to your wrangler.toml. Refer to https://developers.cloudflare.com/workers/learning/using-durable-objects/#durable-object-migrations-in-wranglertoml for more details."
```

Comment:
Intrinsic is a new term that I don't think we use elsewhere, which could add more confusion about how DOs work.
Open to discussion but I think `local` would work, or perhaps more explicitly `you have configured a Durable Object binding that refers to a class defined in this Worker`??

Reply:
I prefer your wording, thanks.